### ISSUE TITLE
[BUG] Function `edges` does not work for graphs of type `SimpleGraph{Integer}`

### ISSUE THREAD

**Issue by simonschoelly** [JuliaGraphs/LightGraphs.jl#1225]
**Description of bug**
The function `edges` does not work for graphs of type `SimpleGraph{Integer}`

**How To Reproduce**
- Create a graph of type `g = SimpleGraph{Integer}`
- run `edges(g)
- run collect on the result.

**Expected behavior**
A vector of eltype `Edge{Integer}` should be returned.

**Actual behavior**
A method errror is thrown.

**Code demonstrating bug**
```
g = SimpleGraph{Intger}(4)
el = edges(g)
collect(el)

ERROR: MethodError: no method matching iterate(::LightGraphs.SimpleGraphs.SimpleEdgeIter{SimpleDiGraph{Integer}}, ::LightGraphs.SimpleGraphs.SimpleEdgeIterState{Int64})
Closest candidates are:
  iterate(::Core.SimpleVector, ::Any) at essentials.jl:568
  iterate(::ExponentialBackOff, ::Any) at error.jl:199
  iterate(::Base.MethodList, ::Any...) at reflection.jl:803
  ...
Stacktrace:
 [1] iterate(::LightGraphs.SimpleGraphs.SimpleEdgeIter{SimpleDiGraph{Integer}}) at /home/simon/.julia/dev/LightGraphs/src/SimpleGraphs/simpleedgeiter.jl:67
 [2] copyto!(::Array{LightGraphs.SimpleGraphs.SimpleEdge{Integer},1}, ::LightGraphs.SimpleGraphs.SimpleEdgeIter{SimpleDiGraph{Integer}}) at ./abstractarray.jl:668
 [3] _collect(::UnitRange{Int64}, ::LightGraphs.SimpleGraphs.SimpleEdgeIter{SimpleDiGraph{Integer}}, ::Base.HasEltype, ::Base.HasLength) at ./array.jl:550
 [4] collect(::LightGraphs.SimpleGraphs.SimpleEdgeIter{SimpleDiGraph{Integer}}) at ./array.jl:544
 [5] |>(::LightGraphs.SimpleGraphs.SimpleEdgeIter{SimpleDiGraph{Integer}}, ::typeof(collect)) at ./operators.jl:813
 [6] top-level scope at none:0
```

**Version information**
 - output from `versioninfo()`
Julia Version 1.1.0
Commit 80516ca202 (2019-01-21 21:24 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = gvim
 - output from `] stats LightGraphs`
(v1.1) pkg> status LightGraphs
    Status `~/.julia/environments/v1.1/Project.toml`
  [ec485272] ArnoldiMethod v0.0.4
  [864edb3b] DataStructures v0.15.0
  [093fc24a] LightGraphs v1.2.0 [`~/.julia/dev/LightGraphs`]
  [699a6c99] SimpleTraits v0.8.0

**Additional context**
Maybe graphs should only be defined for concrete integer types.

**Comment by sbromberger** [comment]
> Maybe graphs should only be defined for concrete integer types.

This, exactly. Mind creating a PR?

```
julia> isabstracttype(Integer)
true

julia> isabstracttype(Int)
false

julia> isconcretetype(Integer)
false

julia> isconcretetype(Int)
true
```

**Comment by simonschoelly** [comment]
Should be easy to fix, will try to do it tonight.

**Comment by simonschoelly** [comment]
Turns out, it is not completely trivial, because at the moment
```

julia> SimpleGraphFromIterator( [Edge(1, 2), Edge(Int8(1), Int8(2)) ])
{2, 1} undirected simple Signed graph
```

I will have to come up with some solution here.

**Comment by sbromberger** [comment]
I think this has been fixed by #1226.
